gpii/node_modules/flowManager/src/FlowManagerUtilities.js

```diff
@@ -43,35 +42,33 @@
         });
     };
 
+    gpii.flowManager.loadSolutionsRegistry = function (flowManager) {
+        var resolvedFolder = fluid.module.resolvePath(flowManager.options.solutionsRegistrySourceDir);
+        flowManager.options.fullSolutionsRegistry = require(resolvedFolder);
```

Comment:
To clarify - I think the best result would be if it was returned to being a dataSource, but one which when given an empty argument to "get", returns the full registry. It looks like all the algorithms we have now are simplest if they are working on the full registry. In future, once it becomes very large, we'll support more forms of "query" to it - for example, the ability to query it by preferences path or whatever.